lib/phoenix/live_dashboard/pages/processes_page.ex

```diff
@@ -35,12 +35,13 @@ defmodule Phoenix.LiveDashboard.ProcessesPage do
   end
 
   defp fetch_processes(params, node, state) do
-    %{search: search, sort_by: sort_by, sort_dir: sort_dir, limit: limit} = params
+    %{search: search, sort_by: sort_by, sort_dir: sort_dir, limit: limit, filter: filter} = params
 
-    {processes, count, state} =
-      SystemInfo.fetch_processes(node, search, sort_by, sort_dir, limit, state)
+    {active_filter, available_filters, processes, count, state} =
+      SystemInfo.fetch_processes(node, filter, search, sort_by, sort_dir, limit, state)
 
-    {processes, count, state}
+    {processes, count,
+     state |> Map.merge(%{filter: active_filter, filter_list: available_filters})}
```

Comment:
Let's change this to return a 5 element tuple, instead of changing the state (which is not necessarily guaranteed to be a map).